Trippy/API/NetworkManager.m

```diff
@@ -0,0 +1,77 @@
+//
+//  NetworkManager.m
+//  Trippy
+//
+//  Created by Catherine Lu on 7/27/22.
+//
+
+#import "NetworkManager.h"
+#import "Reachability.h"
+#import "CoreDataHandler.h"
+#import "CacheDataHandler.h"
+#import "Itinerary.h"
+#import "LocationCollection.h"
+
+@interface NetworkManager ()
+@property (strong, nonatomic) CacheDataHandler *parseHandler;
+@end
+
+@implementation NetworkManager
+
++ (NetworkManager *)shared {
+    static NetworkManager *_sharedManager = nil;
+    
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _sharedManager = [[self alloc] init];
+    });
+
+    return _sharedManager;
+}
+
+- (void)beginNotifier {
+    self.parseHandler = [[CacheDataHandler alloc] init];
+    self.internetReachable = [Reachability reachabilityWithHostname:@"www.google.com"];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(reachabilityChanged)
+                                                 name:kReachabilityChangedNotification
+                                               object:self.internetReachable];
+    [self.internetReachable startNotifier];
+    self.isConnected = [self.internetReachable isReachable];
+    self.isSynced = YES;
```

Comment:
Is this a good default? What happens if you don't have a connection right away, does `reachabilityChanged` fire?

Reply:
I believe it does, but changed isSynced default just in case!